src/server/frontend_xwayland/xwayland_server.cpp

```diff
@@ -64,7 +64,12 @@ mf::XWaylandServer::~XWaylandServer()
       if (xserver_status == RUNNING)
         wm->destroy();
 
-      kill(pid, SIGTERM);
+      if (kill(pid, SIGTERM) == 0)
+      {
+          usleep(100000);           // After 100ms...
+          if (kill(pid, 0) == 0)    // ...if Xwayland is still running...
+            kill(pid, SIGKILL);     // ...then kill it!
+      }
```

Comment:
Upon failure of `kill()`, shouldn't we at least log an error?

Reply:
If the process has already exited, we'll get an "error", but why would we need to log it?

Reply:
I see, never mind